libraries/AP_HAL/utility/Socket.cpp

```diff
@@ -110,6 +110,14 @@ void SocketAPM::set_blocking(bool blocking)
     }
 }
 
+/*
+  set cloexec state
+ */
+void SocketAPM::set_cloexec()
+{
+    fcntl(fd, F_SETFD, FD_CLOEXEC);
```

Comment:
héhéhéh, nope ! same error as me ! SOCK_CLOEXEC is only available on Linux plateform, so that prevent use of SITL on mac and windows. 
But I agree that should be default on all SITL socket

Reply:
As @khancyr says, we need to use fcntl, but I agree with @lucasdemarchi that all sockets should have this set - what would be the reason to not do it?

Reply:
@khancyr SOCK_CLOEXEC exists for a reason. If you are ok with the racy alternative in other platforms like you are doing here, please keep it only on these other platforms. I.e.: change the Socket() constructor to do the fcntl() call itsef.